sys/dev/ti/if_ti.c

```diff
@@ -592,12 +594,12 @@ ti_copy_mem(struct ti_softc *sc, uint32_t tigon_addr, uint32_t len,
 					    sc->ti_membuf2, segsize,
 					    TI_SWAP_NTOH);
 					TI_UNLOCK(sc);
-					bcopy(&sc->ti_membuf2[segresid], ptr,
+					bcopy(&sc->ti_membuf2[segresid], (void *)(__cheri_addr uintptr_t)ptr,
 					    segsize - segresid);
 					TI_LOCK(sc);
 					first_pass = 0;
 				} else
-					ti_bcopy_swap(sc->ti_membuf, ptr,
+					ti_bcopy_swap(sc->ti_membuf, (void *)(__cheri_addr uintptr_t)ptr,
```

Comment:
```suggestion
					ti_bcopy_swap(sc->ti_membuf, (__cheri_fromcap void *)ptr,
```

Reply:
Same question here. I know in purecap __cheri_fromcap will do the right thing since it just returns the capability unchanged but in hybrid it will be relative to kernel ddc.